publishing-statistics/dashboards.qmd

```diff
@@ -583,6 +583,49 @@ See our [Git](../learning-development/git.html#storing-secure-variables) page fo
+</div>
+
+---
+
+### Updating data in a dashboard
+
+---
+
+As mentioned in the [public dashboards section](#public-dashboards), a public dashboard should not be updated with unpublished data until the data is published. However, it is possible to clone the repo and run it locally with unpublished data for testing purposes (see our [guidance on testing with unpublished data](#testing-with-unpublished-data)). This guidance applies to both adding real data to a dashboard that previously used dummy data, and to adding updated data to an existing dashboard.
+
+On the day prior to publication:
+
+1. Create a local branch which you will use to update the data. To do this open your local version of the repo, navigate to the Git window, ensure the current branch is main (see 1) as this is the code producing the current live version of the app. If you're happy the current branch is main, then do a pull to ensure it is up-to-date (see 2). Then click the new branch button (see 3).
+
+![](../images/create_new_branch.png)
+
+2. Give your branch a descriptive name, for example, for a data update for publication on a given date, incorporate the date into the branch name (i.e. data_update_01Jan_2023). Click create.
+3. You can now add your unpublished data files to the data folder in your local copy of the repository (in the file explorer). If you have already added the data using .gitignore as described in the [guidance on testing with unpublished data](#testing-with-unpublished-data), you can now remove the file name from the .gitignore file. You can also remove any dummy data files or old data files from the folder.
+4. Now open the script that reads in data for your dashboard and edit the file paths to point at the new data files.
+5. run tidy_code() and run_tests_locally() (see [guidance on UI tests](#ui-tests)). If changes are found when running the tests locally, make sure you look through these differences and understand them. If you are happy with the changes found in the tests (i.e. they are expected due to your updates), update the .json files. These are tested via GitHub actions every time you do a pull request, so you should always run them in advance of any pull request.
+6. You can run your app to test that it is working, but **do not** commit or push this branch yet.
+
+On the day of publication:
+
+7. At 9:30 on publication day, commit and push the changes in your new branch to GitHub. This makes the data files publicly available via GitHub and so should not be done until 9:30 on the day of publication, however this step does not update the dashboard.
+8.	On the GitHub repo, you should now see a prompt to open a pull request. You should do this, following the template, ticking the boxes to show you have completed tests locally as required and provide details of the changes (i.e. that you've updated to include the most recent data, including dates).
```

Comment:
Should there be less space between the number and the start of the text on this line?

Reply:
Fixed in latest commit